### ISSUE TITLE
Error after configuration, Buffer write error.

### ISSUE THREAD

**Issue by DavidSack** [lavinir/hassio-onedrive-backup#4]
Looked like setup ran successfully but I am seeing this in my logs:
10/27/2022 15:45:38 Info: Backup interval configured to every 24 hours
10/27/2022 15:45:38 Info: Checking backups
10/27/2022 15:45:39 Info: Retrieving existing online backups...
10/27/2022 15:45:39 Info: Found 1 backups to upload.
10/27/2022 15:45:39 Info: Uploading hass_backup (10/27/2022 14:58:57)
10/27/2022 15:45:39 Info: Fetching Local Backup (Slug:90aa5b46)
10/27/2022 15:45:39 Error: Unexpected error. System.Net.Http.HttpRequestException: **Cannot write more bytes to the buffer than the configured maximum buffer size: 2147483647.**
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at hassio_onedrive_backup.Hass.HassioClient.DownloadBackup(String backupSlug) in /src/Hass/HassioClient.cs:line 117
   at hassio_onedrive_backup.Hass.BackupManager.PerformBackupsAsync() in /src/Hass/BackupManager.cs:line 78
   at hassio_onedrive_backup.Program.Main(String[] args) in /src/Program.cs:line 41

Any ideas?

Thanks
Dave

**Comment by uSlackr** [comment]
Same issue.  Here's my log entry - repeats every hour
Running latest HA (22.10.5) on latest supervisor.
```
10/27/2022 17:38:21 Info: Checking backups
10/27/2022 17:38:22 Info: Retrieving existing online backups...
10/27/2022 17:38:22 Info: Found 1 backups to upload.
10/27/2022 17:38:22 Info: Uploading hass_backup (10/27/2022 15:36:27)
10/27/2022 17:38:22 Info: Fetching Local Backup (Slug:162d6859)
10/27/2022 17:38:22 Error: Unexpected error. System.Net.Http.HttpRequestException: Cannot write more bytes to the buffer than the configured maximum buffer size: 2147483647.
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at hassio_onedrive_backup.Hass.HassioClient.DownloadBackup(String backupSlug) in /src/Hass/HassioClient.cs:line 117
   at hassio_onedrive_backup.Hass.BackupManager.PerformBackupsAsync() in /src/Hass/BackupManager.cs:line 78
   at hassio_onedrive_backup.Program.Main(String[] args) in /src/Program.cs:line 41
```

**Comment by lavinir** [comment]
@uSlackr  , @DavidSack , I believe this should only happen when the backup file exceeds 2GB. 
Do you have backups that size ?

**Comment by beastie29a** [comment]
@lavinir, I'm having the same issue and yes my backups are 2.5GB.

**Comment by beastie29a** [comment]
Here's a patch I created that seems to have gotten backups going:
```
diff --git a/onedrive-backup/Hass/HassioClient.cs b/onedrive-backup/Hass/HassioClient.cs
index 8a5081b..fe941cc 100644
--- a/onedrive-backup/Hass/HassioClient.cs
+++ b/onedrive-backup/Hass/HassioClient.cs
@@ -114,10 +114,9 @@ namespace hassio_onedrive_backup.Hass
             ConsoleLogger.LogInfo($"Fetching Local Backup (Slug:{backupSlug})");
             Uri uri = new Uri(Supervisor_Base_Uri_Str + $"/backups/{backupSlug}/download");
             var fileInfo = new FileInfo($"{backupSlug}.tar");
-            var response = await _httpClient.GetAsync(uri);
+            var response = await _httpClient.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead);
             await using var memStream = await response.Content.ReadAsStreamAsync();
             using var fileStream = System.IO.File.Create(fileInfo.FullName);
-            memStream.Seek(0, SeekOrigin.Begin);
             await memStream.CopyToAsync(fileStream);
             ConsoleLogger.LogInfo($"Backup ({backupSlug}) fetched successfully");
             return fileInfo.FullName;
```

Log output:
```
10/28/2022 05:30:41 Info: Backup interval configured to every 72 hours
10/28/2022 05:30:41 Info: Checking backups
10/28/2022 05:30:42 Info: Retrieving existing online backups...
10/28/2022 05:30:43 Info: Found 1 backups to upload.
10/28/2022 05:30:43 Info: Uploading hass_backup (10/27/2022 19:08:51)
10/28/2022 05:30:43 Info: Fetching Local Backup (Slug:0bd83226)
10/28/2022 05:31:04 Info: Backup (0bd83226) fetched successfully
10/28/2022 05:31:05 Info: Starting file upload. (Size:2676439040 bytes. Attempt: 1/3)
10/28/2022 05:32:54 Info: Uploaded 10%
10/28/2022 05:35:15 Info: Uploaded 20%
10/28/2022 05:37:47 Info: Uploaded 30%
10/28/2022 05:39:58 Info: Uploaded 40%
10/28/2022 05:42:07 Info: Uploaded 50%
10/28/2022 05:44:01 Info: Uploaded 60%
10/28/2022 05:45:47 Info: Uploaded 70%
10/28/2022 05:47:34 Info: Uploaded 80%
10/28/2022 05:49:24 Info: Uploaded 90%
10/28/2022 05:51:24 Info: Uploaded 100%
10/28/2022 05:51:31 Info: Upload completed successfully
10/28/2022 05:51:32 Info: Interval Completed.
```

**Comment by konradwalsh** [comment]
Here to say I am having the same.. my backups are 8gb

**Comment by user45876** [comment]
One more, yes, more than 2GB backups. Also wanted to say thank you for making this :)

**Comment by DavidSack** [comment]
Yes I do.

Thanks
Dave

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by lavinir** [comment]
Will add a fix for this in the upcoming update

**Comment by DavidSack** [comment]
Great, thanks!

Sent from Surface Duo
________________________________

**Comment by Rolzzz** [comment]
following so I can also get update for issue

**Comment by lavinir** [comment]
Fixed in V1.1 (hopefully :))
